Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.SealDirectory.cs

```diff
@@ -79,16 +98,50 @@ private static EvaluationResult ComposeSharedOpaqueDirectories(Context context,
         }
         
 
-        private static EvaluationResult SealSourceDirectory(Context context, ModuleLiteral env, EvaluationStackFrame args)
+        private EvaluationResult SealSourceDirectory(Context context, ModuleLiteral env, EvaluationStackFrame args)
         {
-            AbsolutePath path = Args.AsPath(args, 0, false);
-            var optionAsEnumValue = Args.AsNumberOrEnumValueOptional(args, 1);
-            var option = optionAsEnumValue.HasValue ? (SealSourceDirectoryOption)optionAsEnumValue.Value : SealSourceDirectoryOption.TopDirectoryOnly;
-            var tags = Args.AsStringArrayOptional(args, 2);
-            var description = Args.AsStringOptional(args, 3);
-            var patterns = Args.AsStringArrayOptional(args, 4);
+            AbsolutePath path;
+            SealDirectoryKind sealDirectoryKind;
+            string[] tags;
+            string description;
+            string[] patterns;
 
-            var sealDirectoryKind = option == SealSourceDirectoryOption.AllDirectories ? SealDirectoryKind.SourceAllDirectories : SealDirectoryKind.SourceTopDirectoryOnly;
+            if (args.Length > 0 && args[0].Value is ObjectLiteral)
+            {
+                var obj = Args.AsObjectLiteral(args, 0);
+                var directory = Converter.ExtractDirectory(obj, m_sealRoot, allowUndefined: false);
+                path = directory.Path;
+                var include = Converter.ExtractString(obj, m_sealInclude, allowUndefined: true);

```

Comment:
>ExtractString [](start = 40, length = 13)

There is an 'ExtractStringLiteral' that may suit you better

Reply:
Thanks!